backend/fms_core/services/container.py

```diff
@@ -0,0 +1,50 @@
+from datetime import datetime
+from django.core.exceptions import ValidationError
+from fms_core.models import Container
+
+def get_container(barcode=None):
+    container = None
+    errors = []
+    warnings = []
+
+    try:
+        container = Container.objects.get(barcode=barcode)
```

Comment:
We should differentiate the error message when a barcode is missing (None) than when there is no DB entry for a barcode.

Reply:
This is not just true here ... It could go a long way in identifying the issues. In some case, barcode may be optional but in those case this function should not be called at all. If it is called, we want to know.

Reply:
I suggest renaming this function get_container_from_barcode so it makes it more obvious that the Barcode value is required for this function

Reply:
I really wanted to make it a convention in Services to have defaults for all the params passed to those functions. Or we could have no default for all of them, but then there is some refactor to do here and there.

I don't really understand what you dont like here? If you try to find a Container without barcode, then it still doesnt exist

Reply:
Right I'll add make it a conditional statement that appends an error if the barcode value is None.

I understand your point regarding having a default value that is not a valid case, but at the same time I don't want to start to manage all the params default values in the code, so I'll leave it as it is for now, with the suggestion we agreed on.